src/subscription/subscription.controller.ts

```diff
@@ -58,34 +58,40 @@ export class SubscriptionController {
         @Res() response: Response,
         @Query() query: { unsubscribeReference?: string },
     ): Promise<void> {
-        const result = await this.subscriptionService.deleteBySubAndGrantId(
+        const ref = query?.unsubscribeReference;
+        let result = await this.subscriptionService.deleteBySubAndGrantId(
             id,
             grantId,
         );
-        if (query?.unsubscribeReference) {
+        if (result.affected == 0) {
+            result = await this.subscriptionService.deleteByEmailAndGrantId(
+                id,
+                grantId,
+            );
+        }
+
+        if (ref && ref !== 'undefined') {
```

Comment:
fixed the refs being passed as string `undefined` values from the front-end in my pr btw